observability-lib/grafana/dashboard.go

```diff
@@ -12,22 +16,150 @@ const (
 	TypePlatformDocker     TypePlatform = "docker"
 )
 
-type DashboardOptions struct {
-	Name              string
-	FolderUID         string
-	Platform          TypePlatform
-	OCRVersion        string
-	MetricsDataSource *DataSource
-	LogsDataSource    *DataSource
-	AlertsTags        map[string]string
-	SlackWebhookURL   string
-	SlackToken        string
-	SlackChannel      string
-}
-
 type Dashboard struct {
 	Dashboard            *dashboard.Dashboard
 	Alerts               []alerting.Rule
 	ContactPoints        []alerting.ContactPoint
 	NotificationPolicies []alerting.NotificationPolicy
 }
+
+func (db *Dashboard) GenerateJSON() ([]byte, error) {
+	dashboardJSON, err := json.MarshalIndent(db, "", "  ")
+	if err != nil {
+		return nil, err
+	}
+
+	return dashboardJSON, nil
+}
+
+type DeployOptions struct {
+	GrafanaURL            string
+	GrafanaToken          string
+	FolderName            string
+	EnableAlerts          bool
+	NotificationTemplates string
+}
+
+func (db *Dashboard) DeployToGrafana(options *DeployOptions) error {
+	grafanaClient := api.NewClient(
+		options.GrafanaURL,
+		options.GrafanaToken,
+	)
+
+	folder, errFolder := grafanaClient.FindOrCreateFolder(options.FolderName)
+	if errFolder != nil {
+		return errFolder
+	}
+
+	newDashboard, _, errPostDashboard := grafanaClient.PostDashboard(api.PostDashboardRequest{
+		Dashboard: db.Dashboard,
+		Overwrite: true,
+		FolderID:  int(folder.ID),
+	})
+	if errPostDashboard != nil {
+		return errPostDashboard
+	}
+
+	// Create alerts for the dashboard
+	if options.EnableAlerts && db.Alerts != nil && len(db.Alerts) > 0 {
+		// Get alert rules for the dashboard
+		alertsRule, errGetAlertRules := grafanaClient.GetAlertRulesByDashboardUID(*newDashboard.UID)
+		if errGetAlertRules != nil {
+			return errGetAlertRules
+		}
+
+		// delete alert rules for the dashboard
+		for _, rule := range alertsRule {
+			_, _, errDeleteAlertRule := grafanaClient.DeleteAlertRule(*rule.Uid)
+			if errDeleteAlertRule != nil {
+				return errDeleteAlertRule
+			}
+		}
+
+		// Create alert rules for the dashboard
```

Comment:
how do the alerts defined here play with the alerts we define in the alerts repo?

Reply:
Those alerts are Grafana managed alert rules
<img width="1377" alt="image" src="https://github.com/user-attachments/assets/b1079927-35fc-469e-92c7-4a86c546ae9c">
The alerts in alerts repo are data source-managed alert rule